src/customnodesession.cpp

```diff
@@ -0,0 +1,202 @@
+    // In this case shared library is responsible for cleaning up resources (memory).
+    if (result != 0) {
+        SPDLOG_LOGGER_ERROR(dag_executor_logger, "Node {}; session: {}; has failed custom node execution with return code: {}", getName(), getSessionKey(), result);
+        notifyEndQueue.push({node, getSessionKey()});
+        return StatusCode::NODE_LIBRARY_EXECUTION_FAILED;
+    }
+
+    if (outputTensors == nullptr) {
+        SPDLOG_LOGGER_ERROR(dag_executor_logger, "Node {}; session: {}; has corrupted outputs handle", getName(), getSessionKey());
+        notifyEndQueue.push({node, getSessionKey()});
+        return StatusCode::NODE_LIBRARY_OUTPUTS_CORRUPTED;
+    }
+
+    if (outputTensorsLength <= 0) {
+        SPDLOG_LOGGER_ERROR(dag_executor_logger, "Node {}; session: {}; has corrupted number of outputs", getName(), getSessionKey());
+        // Cleanup whatever is possible
+        this->library.releaseTensors(outputTensors);
+        notifyEndQueue.push({node, getSessionKey()});
+        return StatusCode::NODE_LIBRARY_OUTPUTS_CORRUPTED;
+    }
+
+    // At this point this is important we do not exit before finishing the loop.
+    // There will be memory leak if any tensor is not converted into blob.
+    // Blob destructor is responsible for cleaning up resources.
+    Status status = StatusCode::OK;
+    for (int i = 0; i < outputTensorsLength; i++) {
+        InferenceEngine::Blob::Ptr resultBlob;
+        auto result = this->createBlob(&outputTensors[i], resultBlob);
+        if (!result.ok()) {
+            SPDLOG_LOGGER_ERROR(dag_executor_logger, "Node {}; session: {}; has corrupted output: {}; cannot convert to blob", getName(), getSessionKey(), outputTensors[i].name);
+            // Cleanup whatever is possible
+            this->library.releaseBuffer(&outputTensors[i]);
+            status = StatusCode::NODE_LIBRARY_OUTPUTS_CORRUPTED;
+            continue;
+        }
+        this->resultBlobs.emplace(std::make_pair(std::string(outputTensors[i].name), std::move(resultBlob)));
+    }
+
+    this->library.releaseTensors(outputTensors);
+
+    notifyEndQueue.push({node, getSessionKey()});
+    return status;
+}
+
+Status CustomNodeSession::fetchResult(const std::string& name, InferenceEngine::Blob::Ptr& resultBlob) {
+    auto it = resultBlobs.find(name);
+    if (it == resultBlobs.end()) {
+        return StatusCode::NODE_LIBRARY_MISSING_OUTPUT;
+    }
+    resultBlob = it->second;
+    return StatusCode::OK;
+}
+
+Status CustomNodeSession::createBlob(const struct CustomNodeTensor* tensor, InferenceEngine::Blob::Ptr& resultBlob) {
+    InferenceEngine::TensorDesc desc;
+
+    // precision
+    InferenceEngine::Precision precision = toInferenceEnginePrecision(tensor->precision);
+    if (precision == InferenceEngine::Precision::UNSPECIFIED) {
+        SPDLOG_LOGGER_ERROR(dag_executor_logger, "Node {}; session: {}; Unspecified output precision from custom node",
```

Comment:
DEBUG

Reply:
Why should it be DEBUG? In my opinion this is something administrator should see - using shared library that yields outputs with incorrect metadata.